service/frontend/dcRedirectionHandler.go

```diff
@@ -546,16 +587,37 @@ func (handler *DCRedirectionHandlerImpl) SignalWithStartWorkflowExecution(
 	request *shared.SignalWithStartWorkflowExecutionRequest,
 ) (*shared.StartWorkflowExecutionResponse, error) {
 
-	targetDC, err := handler.redirectionPolicy.GetTargetDatacenterByName(request.GetDomain())
+	targetDC, err := handler.redirectionPolicy.GetTargetDataCenterByName(request.GetDomain())
 	if err != nil {
 		return nil, err
 	}
 
-	if targetDC == handler.currentClusteName {
-		return handler.frontendHandler.SignalWithStartWorkflowExecution(ctx, request)
+	var ok bool
+	var resp *shared.StartWorkflowExecutionResponse
+	attempt := APIAttemptCount
```

Comment:
APIAttemptCount and DomainActivenessAttemptCount looks weird to me. Basically you want to use these two "magic" number to help decide which handler API to use.

My personal suggestion is to remove these two number to make code looks cleaner, and another way to organize your code:

```
var handlers []Handler
if targetDC == handler.currentClusterName {
    handlers :=[]Handler{handler.frontendHandler, 
    handler.service.GetClientBean().GetRemoteFrontendClient(targetDC))
else{
    handlers :=[]Handler{handler.service.GetClientBean().GetRemoteFrontendClient(targetDC),
    handler.frontendHandler)
}

resp, err = handler[0].XXXX(ctx, request)
_, ok = handler.isDomainNotActiveError(err) 
if ok{
    resp, err = handler[1].XXXX(ctx, request)
}   
return resp, err
}